portfolio/src/main/webapp/script.js

```diff
@@ -106,32 +104,105 @@ function createCommentElement(comment) {
 
   return commentElement;
 }
-async function displayComments() {
+document.addEventListener('DOMContentLoaded', () => {
+  document.getElementById('maxNum').addEventListener('input', () => {
+    let parentElement = document.getElementById('comments');
+    while (parentElement.firstChild) {
+      parentElement.removeChild(parentElement.lastChild);
+    }
+    cursorStr = null;
+    displayComments(cursorStr);
+  });
+});
+
+let cursorStr = null;
+async function displayComments(cursorString) {
   maxNumberOfComments = document.getElementById('maxNum').value;
-  const params = new URLSearchParams();
-  params.append('max_num_of_contents', maxNumberOfComments);
-  const response = await fetch(`/data?max_num_of_comments=${maxNumberOfComments}`);
-  const comments = await response.json();
 
-  commentElement = document.getElementById('comments');
-  while (commentElement.lastElementChild) {
-    commentElement.removeChild(commentElement.lastElementChild);
-  }
+  const response = await fetch(`/data?max_num_of_comments=${maxNumberOfComments}&cursor=${cursorStr}`);
+  const commentData = await response.json();
 
   const commentListElement = document.getElementById('comments');
-  comments.forEach((comment) => {
-    commentListElement.appendChild(createCommentElement(comment));
+  cursorStr = commentData['cursorString'];
+
+  let commentList= commentListElement.children;
+
+  for (let i = 0; i < commentList.length; i++) {
+    commentList[i].style.display = "none";
+  }
+
+  if (commentData['commentList'].length < maxNumberOfComments) {
+    document.getElementById('next-comments').style.display = "none";
+    document.getElementById('end-of-comments-sign').style.display = "block";
+  }
+
+  let commentGroup = document.createElement('div');
+  let numberOfCommentGroups = commentListElement.childElementCount;
+  numberOfCommentGroups++;
+  commentGroup.id = numberOfCommentGroups.toString();
+  commentGroup.style.display = "block";
+  commentData['commentList'].forEach((comment) => {
+    commentGroup.appendChild(createCommentElement(comment));
   });
+  commentListElement.appendChild(commentGroup);
 }
 
+document.addEventListener('DOMContentLoaded', () => {
+  let lastPage = 0;
+  document.getElementById('next-comments').addEventListener('click', (cursorStr) => {
+    document.getElementById('prev-comments').style.display = "block";
+    let commentList = document.getElementById('comments').children;
+
+    let i;
+    let needToSendNextRequest = true;
+    for (i = 0; i < commentList.length; i++) {
```

Comment:
instead of this, can you keep another index

shownCommentIndex : number | null

if shownCommentIndex is non-null then you know that that was the index that was last shown!  You can then index into commentList directly; ie:

commentList[this.shownCommentIndex].classList.add('hidden');
if (this.shownCommentIndex+1 < commentList.length) {
  this.shownCommentIndex++;
  commentList[shownCommentIndex].classList.add('hidden');
} else {
   this.lastPageRequested++;
   requestAndDisplayNewPage(cursorStr);  // note i renamed your function
}

same for prev